### PR TITLE
[storage] change ledger_info key from Verison to epoch_num

### PR DESCRIPTION
**Pull request by lightmark**
## Motivation

we will use mostly epoch number to query `ledger_info`, needed by consensus reconfig.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

CI

### CODE REVIEW

File: storage/libradb/src/ledger_store/ledger_info_test.rs [hunk 1]
```diff
@@ -29,11 +29,11 @@ prop_compose! {
                 let ledger_info = p.ledger_info();
                 LedgerInfoWithSignatures::new(
                     LedgerInfo::new(
-                        start_version + i as u64,
+                        start_epoch + i as Version,
```
Comment (zamsden) [inline storage/libradb/src/ledger_store/ledger_info_test.rs]: `as Version` ?
billo

Comment (lightmark) [reply storage/libradb/src/ledger_store/ledger_info_test.rs]: The first param is `version` so I just assign `epoch_num` as `version`

Comment (zamsden) [reply storage/libradb/src/ledger_store/ledger_info_test.rs]: Why not pass both start_epoch and start_version ?

Comment (lightmark) [reply storage/libradb/src/ledger_store/ledger_info_test.rs]: The version cannot be random. If version_a < version_b then epoch_a must be < apoch_b. Assign proper random versions don't help what we are testing here...

**Pull request merged**